### DISCUSSION TITLE
How to add layer to headreco-generated mesh?

### DISCUSSION THREAD

**Discussion by ramBrain** [simnibs/simnibs#155] [7.Creating head models]
Hi everyone,

I generated a mesh using `headreco`, and I converted the output to the SimNIBS4 format using `convert_3_to_4`. However, the `label_prep` folder and `tissue_labeling_upsampled.nii.gz` are not generated by the conversion process, hence I am not be able to use `add_tissue_to_upsampled`. Any workaround for this?

Thank you,
Ramtin

**Comment by jduemose** [comment]
Hi Ramtin

Sorry about the late answer. You are correct that `tissue_labeling_upsampled.nii.gz` is only available for meshes generated with `charm`. With `charm`, the labeling image is meshed directly (no surfaces) so it is easy to just edit this. However, this is not the case for `headreco` which constructs nested (non-intersecting) surfaces in between which tetrahedra are then filled.

Adding extra tissues to a `headreco` run is therefore not straightforward. I would *imagine* that if you have an STL file describing the surface of whatever tissue you want to add (and it doesn't intersect with any of the other tissues), you could perhaps add this to the `.geo` file which sets up the meshing for `gmsh`. You would then place this STL file in the same folder as the others. But I don't know if this is a possibility for you?

**Reply by ramBrain** [reply]
Hi,

The tissue I want to add is simply a volume (NIFTI) file. Being a lesion, it will necessarily intersect the other tissues.

I wonder if to your knowledge SPM12 (or any other toolbox) can be used to this purpose by feeding it with the headreco generated mesh?

Thanks,
Ramtin

**Reply by jduemose** [reply]
I don't think that will work unfortunately as the operations is hardcoded on different tissues. Perhaps I may suggest a different approach (as I assume this is related to #147 ?): I am just about to try to write a small command line tool which will allow someone to export a leadfield *calculated with simnibs* to brainstorm; that is, use simnibs for head model generation *and* leadfield generation. I have done similar with fieldtrip and mne-python. I don't know how tricky it is going to be or exactly how long it will take (but I aim no more than a month or so). So, unless you are in a hurry, that might be worth trying out? Perhaps/hopefully simnibs will be able to calculate leadfields for its own mesh. This will allow you to use the CHARM output which is more easy to modify (I am not sure if there are some assumptions applied in the duneuro code that our meshes may not conform with.)
